src/Symfony/Component/Translation/Bridge/Crowdin/CrowdinProviderFactory.php

```diff
@@ -66,16 +67,13 @@ public function create(Dsn $dsn): ProviderInterface
         $host = 'default' === $dsn->getHost() ? $this->getHost($dsn) : $dsn->getHost();
         $endpoint = sprintf('%s%s', $host, $dsn->getPort() ? ':'.$dsn->getPort() : '');
 
-        $filesDownloader = $this->client;
-
-        $client = $this->client->withOptions([
-            'base_uri' => 'https://'.$endpoint,
+        $client = ScopingHttpClient::forBaseUri($this->client, sprintf('https://%s/projects/%d/', $endpoint, $this->getUser($dsn)), [
```

Comment:
There is one API method that not requires the `projects/%d` prefix - Add Storage. Probably with these changes the 'Add Storage' request will be broken.

Reply:
`ScopingHttpClient` adds the bearer conditionally, and the URL to download the files is on another domain, so this should be handled.

Reply:
I mean the [Add Storage](https://support.crowdin.com/api/v2/#operation/api.storages.post) API method, that does not contain project ID (`{protocol}://{host}/api/v2/storages`). In this case, the Provider will try to call this method with the wrong base URL, isn't it?

Reply:
The base URI is used only when the requested URL is a relative one.
But when downloading from S3, a full URL is provided.

Reply:
Got it, regarding downloading from S3 now I understand. But I mean regular API calls. Not all of them containing `projects/%d` in base URI. In this case, $client will be pre-configured with `https://%s/projects/%d/` base URI and requests will fail for the 'Add Storage' method as it does not contain the `projects` prefix.